std/functional.d

```diff
@@ -104,31 +104,7 @@ the parameter or provide the symbol via the $(D parmName) argument.
 If $(D fun) is not a string, $(D unaryFun) aliases itself away to $(D fun).
 */
 
-template unaryFun(alias fun, string parmName = "a")
-{
-    static if (is(typeof(fun) : string))
-    {
-        static if (!fun._ctfeMatchUnary(parmName))
-        {
-            import std.traits, std.typecons, std.meta;
-            import std.algorithm, std.conv, std.exception, std.math, std.range, std.string;
-        }
-        auto unaryFun(ElementType)(auto ref ElementType __a)
-        {
-            mixin("alias " ~ parmName ~ " = __a ;");
-            return mixin(fun);
-        }
-    }
-    else static if (needOpCallAlias!fun)
-    {
-        // Issue 9906
-        alias unaryFun = fun.opCall;
-    }
-    else
-    {
-        alias unaryFun = fun;
-    }
-}
+alias unaryFun(alias fun, string parmName = "a") = naryFun!(fun, parmName);
```

Comment:
To take full advantage of the CT optimizations to `naryFun`, this should now be:
```D
template unaryFun(alias fun, parmName...)
    if(parmName.length <= 1)
{
    alias unaryFun = naryFun!(fun, 1, parmName);
}
template binaryFun(alias fun, parmNames...)
    if(parmNames.length <= 2)
{
    alias binaryFun = naryFun!(fun, 2, parmNames);
}
```
Otherwise `unaryFun` and `binaryFun` - by far the most common instantiations of `naryFun` for the forseeable future -